From: Klaus Schmidinger <Klaus.Schmidinger <at> tvdr.de>
Subject: Fix for recording problem in VDR 1.7.20
Newsgroups: gmane.linux.vdr
Date: 2011-08-19 16:43:03 GMT (1 day, 15 hours and 15 minutes ago)
There have been some reports about recording problems with VDR 1.7.20
on some HD channels.
This patch should fix this.

Klaus
diff -Naur vdr-1.7.20/remux.c vdr-1.7.20a//remux.c
--- vdr-1.7.20/remux.c	2011-08-21 09:59:15.233878122 +0200
+++ vdr-1.7.20a//remux.c	2011-08-21 10:01:52.553192393 +0200
@@ -1042,8 +1042,10 @@
                                      payloadUnitOfFrame = (payloadUnitOfFrame + 1) % -framesPerPayloadUnit;
                                      if (payloadUnitOfFrame != 0 && independentFrame)
                                         payloadUnitOfFrame = 0;
-                                     if (payloadUnitOfFrame)
+                                     if (payloadUnitOfFrame) {
+                                        newPayload = false;
                                         newFrame = false;
+                                        }
                                      }
                                   if (framesPerPayloadUnit <= 1)
                                      scanning = false;
